LibCore/CProcess.cpp

```diff
@@ -0,0 +1,104 @@
+#include "CProcess.h"
+
+#include <sched.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <signal.h>
+#include <unistd.h>
+
+void handle_sigchld(int)
+{
+    dbgprintf("CProcess(%d) Got SIGCHLD\n", getpid());
+    int pid = waitpid(-1, nullptr, 0);
+    dbgprintf("CProcess(%d) waitpid() returned %d\n", getpid(), pid);
+    ASSERT(pid > 0);
+}
+
+CProcess::CProcess(int pid)
+    : m_pid(pid)
+{
+    static bool signal_handled = false;
+    if (!signal_handled) {
+        signal_handled = true;
+        signal(SIGCHLD, handle_sigchld);
```

Comment:
I don't love that library code takes control over the `SIGCHLD` signal just because you use `CProcess`.
If there's an existing signal handler for `SIGCHLD`, it's now forgotten and whoever set that up has no idea.
I don't know what the right solution here is, perhaps remembering the old `SIGCHLD` handler and having `CProcess` call that after it's done with his handler?

Reply:
I suppose, though my inclination would be to just pretend it isn't a problem until it is. Chaining has its own set of issues: what if someone else wanted to uninstall their handler later on, and so on?

Reply:
Yeah, fair enough.